File: py_topic_pkg/scripts/cmd_vel_pub.py
#!/usr/bin/env python3

import rospy
from geometry_msgs.msg import Twist

class CmdVelPubNode:

    def __init__(self):
        # Publisher requires 3 paramters
        #  1. topic name
        #  2. topic msg type
        #  3. topic queue size
        self.cmd_vel_pub_ = rospy.Publisher("cmd_vel", Twist, queue_size=10)
        self.timer_ = rospy.Timer(rospy.Duration(1.0/10.0), self.pub_msg)
        self.twist_ = Twist()

    def pub_msg(self, event=None):
        # geometry_msgs.Twist
        # ref: http://docs.ros.org/en/melodic/api/geometry_msgs/html/msg/Twist.html
        self.twist_.linear.x = 0.5
        self.twist_.angular.z = 1.0
        
        self.cmd_vel_pub_.publish(self.twist_)

def cmd_vel_node():
    rospy.init_node('cmd_vel_node', anonymous=True)
    cmd_vel_pub_node = CmdVelPubNode()
    rospy.spin()

if __name__ == '__main__':
    try:
        cmd_vel_node()
    except rospy.ROSInterruptException:
        pass

File: cpp_topic_pkg/src/cmd_vel_pub.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>

class CmdVelPubNode
{
private:
    ros::Publisher cmd_vel_pub_;
    ros::Timer timer_;

    geometry_msgs::Twist twist_msg_;

public:
    CmdVelPubNode(ros::NodeHandle *nh) {
        ROS_INFO("Publisher and Subscriber initialized");
        timer_ = nh->createTimer(ros::Duration(0.1), &CmdVelPubNode::timerCallback, this);
        cmd_vel_pub_ = nh->advertise<geometry_msgs::Twist>("cmd_vel", 10);
    }

    void timerCallback(const ros::TimerEvent& event){

        twist_msg_.linear.x = 0.5;
        twist_msg_.angular.z = 0.5;

        cmd_vel_pub_.publish(twist_msg_);
    }
};

int main(int argv, char** argc) {
    
    ros::init(argv, argc, "cmd_vel_node");
    // ros::NodeHandle nh("my_namespace");
    ros::NodeHandle nh;
    
    CmdVelPubNode cmd_pub_node(&nh);
    
    ros::spin();

    return 0;
}